internal/envoygateway/config/config.go

```diff
@@ -36,19 +38,72 @@ type Server struct {
 	DNSDomain string
 	// Logger is the logr implementation used by Envoy Gateway.
 	Logger logging.Logger
+	// Leader election settings
+	LeaderElection *LeaderElection
+}
+
+type LeaderElection struct {
+	// LeaseDuration defines the time non-leader contenders will wait before attempting to claim leadership. It's based on the timestamp of the last acknowledged signal. The default setting is 15 seconds.
+	LeaseDuration *time.Duration
+	// RenewDeadline represents the time frame within which the current leader will attempt to renew its leadership status before relinquishing its position. The default setting is 10 seconds.
+	RenewDeadline *time.Duration
+	// RetryPeriod denotes the interval at which LeaderElector clients should perform action retries. The default setting is 2 seconds.
+	RetryPeriod *time.Duration
+	// Enables or disables the leader election feature. It is enabled by default.
+	Enabled bool
 }
 
 // New returns a Server with default parameters.
 func New() (*Server, error) {
+	settings, err := withLeaderElectionSettings()
+	if err != nil {
+		return nil, err
+	}
+
 	return &Server{
 		EnvoyGateway: v1alpha1.DefaultEnvoyGateway(),
 		Namespace:    env.Lookup("ENVOY_GATEWAY_NAMESPACE", DefaultNamespace),
 		DNSDomain:    env.Lookup("KUBERNETES_CLUSTER_DOMAIN", DefaultDNSDomain),
 		// the default logger
-		Logger: logging.DefaultLogger(v1alpha1.LogLevelInfo),
+		Logger:         logging.DefaultLogger(v1alpha1.LogLevelInfo),
+		LeaderElection: settings,
 	}, nil
 }
 
+func parseEnvDuration(envVar string, duration **time.Duration) error {
```

Comment:
can we also make `duration` as a return value instead of using secondary pointer? (more like a gopher this way.